client/my-sites/site-settings/jetpack-credentials/credentials-form/index.jsx

```diff
@@ -0,0 +1,261 @@
+/** @format */
+/* eslint-disable */
+/**
+ * External dependendies
+ */
+import React, { Component } from 'react';
+import { get } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormSelect from 'components/forms/form-select';
+import FormTextInput from 'components/forms/form-text-input';
+import FormLabel from 'components/forms/form-label';
+import FormTextArea from 'components/forms/form-textarea';
+import FormInputValidation from 'components/forms/form-input-validation';
+
+export class CredentialsForm extends Component {
+	state = {
+		showPublicKeyField: false,
+		form: {
+			protocol: this.props.protocol,
+			host: this.props.host,
+			port: this.props.port,
+			user: this.props.user,
+			pass: this.props.pass,
+			abspath: this.props.abspath,
+			kpub: this.props.kpub,
+		},
+		formErrors: {
+			host: false,
+			port: false,
+			user: false,
+			pass: false,
+		},
+	};
+
+	componentWillReceiveProps( nextProps ) {
+		const changedFields = [ 'protocol', 'host', 'port', 'user' ].filter(
+			key => nextProps[ key ] !== this.props[ key ]
+		);
+
+		const changedForm = changedFields.reduce(
+			( props, key ) => ( { ...props, [ key ]: nextProps[ key ] } ),
+			{}
+		);
+		const resetErrors = changedFields.reduce(
+			( props, key ) => ( { ...props, [ key ]: false } ),
+			{}
+		);
+
+		this.setState( {
+			form: { ...this.state.form, ...changedForm },
+			formError: { ...this.state.formErrors, resetErrors },
+		} );
+	}
+
+	handleFieldChange = event => {
+		const formState = get( this.state, 'form', {} );
+		const { formErrors } = this.state;
+
+		formState[ event.target.name ] = event.target.value;
+		formErrors[ event.target.name ] = false;
```

Comment:
let's be careful not to directly mutate `this.state` !

Reply:
Yep, this whole function actually wasn't even needed, so I removed it :-)

Reply:
Yep, this whole part wasn't even needed, so I've removed it now.